### PR TITLE
compatibility with jackson 2.6.7 and json-smart 1.3.1

### PR DESCRIPTION
**Pull request by SomkaPe**
https://github.com/AzureAD/microsoft-authentication-library-for-java/issues/240
https://github.com/AzureAD/microsoft-authentication-library-for-java/issues/213

### CODE REVIEW

File: src/integrationtest/java/labapi/LabService.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
     static <T> T convertJsonToObject(final String json, final Class<T> clazz) {
         try {
             return mapper.readValue(json, clazz);
-        } catch (JsonProcessingException e) {
+        } catch (IOException e) {
             throw new RuntimeException("JsonProcessingException: " + e.getMessage(), e);
```
Comment (henrik-me) [inline src/integrationtest/java/labapi/LabService.java:28]: >JsonProcessingException [](start = 40, length = 23)

nit: consider changing this?

File: src/main/java/com/microsoft/aad/msal4j/TokenResponse.java [hunk 2]
```diff
@@ -52,28 +52,28 @@ static TokenResponse parseJsonObject(final JSONObject jsonObject)
...
         String clientInfo = null;
         if (jsonObject.containsKey("client_info")) {
-            clientInfo = jsonObject.getAsString("client_info");
+            clientInfo = JSONObjectUtils.getString(jsonObject, "client_info");
         }
 
         long expiresIn = 0;
         if (jsonObject.containsKey("expires_in")) {
-            expiresIn = jsonObject.getAsNumber("expires_in").longValue();
+            expiresIn = Long.parseLong(jsonObject.getAsString("expires_in"));
```
Comment (henrik-me) [inline src/main/java/com/microsoft/aad/msal4j/TokenResponse.java:71]: >Long.parseLong( [](start = 24, length = 15)

should this have been a mthod in the JSONObjectUtils instead? to align behavior and parsing logic in one place?

Comment (SomkaPe) [reply src/main/java/com/microsoft/aad/msal4j/TokenResponse.java:71]: yes, but that method expects json value as JSON number, so fails if it is string. AAD returns those values as JSON number, B2C as JSON String,  so can not rely on JSONObjectUtils

File: src/integrationtest/java/labapi/LabService.java (hunk 1 above)
Comment (SomkaPe) [reply src/integrationtest/java/labapi/LabService.java:28]: good point - going to change message

**Pull request merged**